core/src/main/java/com/homeaway/streamplatform/streamregistry/resource/ClusterResource.java

```diff
@@ -145,10 +145,14 @@ public Response getAllClusters(@ApiParam(value = "clusterName") @QueryParam("clu
 
             if(type != null && !type.isEmpty()) {
                 clusterList = clusterList.stream()
+                    .filter(e -> e.getClusterKey().getType()!= null)
                     .filter(e -> e.getClusterKey().getType().equalsIgnoreCase(type))
                     .collect(Collectors.toList());
             }
-
+            if (clusterList.isEmpty()) {
+                return Response
+                    .status(Response.Status.NOT_FOUND.getStatusCode()).build();
```

Comment:
I think a `200 OK []` response is fine. 

> if I ask for a set of objects that match a query, a null set is a valid response and I want that returned with a 200 code. The rationale for this is that the query was valid, it succeeded and the query returned nothing.

https://softwareengineering.stackexchange.com/a/203493/279825

Reply:
Plus, is there really a need to continuously collect the stream to a list rather than continuing to filter it just as a `Stream` object?